internal/rootcoord/alter_collection_task.go

```diff
@@ -58,23 +59,16 @@ func (a *alterCollectionTask) Execute(ctx context.Context) error {
 	newColl := oldColl.Clone()
 	updateCollectionProperties(newColl, a.Req.GetProperties())
 
-	ts := a.GetTs()
-	redoTask := newBaseRedoTask(a.core.stepExecutor)
-	redoTask.AddSyncStep(&AlterCollectionStep{
-		baseStep: baseStep{core: a.core},
-		oldColl:  oldColl,
-		newColl:  newColl,
-		ts:       ts,
-	})
+	if err := a.core.ExpireMetaCache(ctx, a.Req.GetDbName(), []string{}, oldColl.CollectionID, "", a.GetTs(), proxyutil.SetMsgType(commonpb.MsgType_AlterCollection)); err != nil {
+		return err
+	}
 
-	a.Req.CollectionID = oldColl.CollectionID
-	redoTask.AddSyncStep(&BroadcastAlteredCollectionStep{
-		baseStep: baseStep{core: a.core},
-		req:      a.Req,
-		core:     a.core,
-	})
+	if err := a.core.meta.AlterCollection(ctx, oldColl, newColl, a.GetTs()); err != nil {
+		return err
+	}
 
-	return redoTask.Execute(ctx)
+	a.Req.CollectionID = oldColl.CollectionID
+	return a.core.broker.BroadcastAlteredCollection(ctx, a.Req)
```

Comment:
I think we should redo `BroadcastAlteredCollection`.

Reply:
there is no difference between use `redoTask.AddSyncStep(&BroadcastAlteredCollectionStep`

Reply:
I mean it shouldn't be a sync step.